R/ResamplingSpCVBuffer.R

```diff
@@ -72,47 +89,55 @@ ResamplingSpCVBuffer = R6Class("ResamplingSpCVBuffer",
     #'   Returns the number of resampling iterations, depending on the
     #'   values stored in the `param_set`.
     iters = function() {
-      as.integer(length(self$instance$fold))
+      as.integer(length(self$instance))
     }
   ),
 
   private = list(
-    .sample = function(ids, coords, crs) {
+    .sample = function(ids, response, coords, positive, crs, properties) {
 
       require_namespaces(c("blockCV", "sf"))
 
-      points = sf::st_as_sf(coords,
+      pars = self$param_set$get_values()
+
+      if (!isTRUE("twoclass" %in% properties) && isTRUE(pars$spDataType == "PB")) {
+        stop("spDataType = 'PB' should only be used with two-class response")
+      }
+
+      if (!is.null(pars$addBG) && isTRUE(pars$spDataType == "PA")) {
+        stop("Parameter addBG should only be used with spDataType = 'PB'")
+      }
+
+      # Recode response to 0/1 for twoclass
+      if (isTRUE("twoclass" %in% properties)) {
+        response = ifelse(response == positive, 1, 0)
+        pars$species = "response"
+      }
+
+      data = sf::st_as_sf(cbind(response, coords),
         coords = c("x", "y"),
         crs = crs)
 
-      inds = blockCV::buffering(
-        speciesData = points,
-        theRange = self$param_set$values$range,
-        progress = FALSE
-      )
+      inds = invoke(blockCV::buffering,
+        speciesData = data,
+        progress = FALSE,
+        .args = pars)
 
-      inds = map(inds$folds, function(x) {
+      map(inds$folds, function(x) {
```

Comment:
We should probably prefix `map` & friends with `mlr3misc::` to be more explicit here.